File: adminweb/src/app/classes/products/ValueConstraint.ts
export class CurrencyBasesValue {
    currencyId: string = "";
    currencyName: string = "";
}

export class ValueConstraint extends CurrencyBasesValue {
    /**
     * Min amount of money.
     */
    minVal: number | null= 0.0;
    /**
     * Max amount of money.
     */
    maxVal: number | null= 0.0;
    /**
     * Default amount of money.
     */
    defaultVal: number | null= 0.0;
}

export class CurrencyLimitValue extends CurrencyBasesValue {
    limit = 0.0;
}

File: common-product/src/main/java/tech/corefinance/product/common/model/ValueConstraint.java
package tech.corefinance.product.common.model;

import lombok.Data;

@Data
public class ValueConstraint {
    private String currencyId;
    private String currencyName;
    /**
     * Min amount of money.
     */
    private Double minVal;
    /**
     * Max amount of money.
     */
    private Double maxVal;
    /**
     * Default amount of money.
     */
    private Double defaultVal;
}
